### PR TITLE
Check both TCP and UDP ports

### PR DESCRIPTION
**Pull request by jcastillocano**
I added a new param (Protocol) to be able to check both TCP and UDP ports using TCPSocket and UDPSocket classes.

I also added some spec tests for check-ports.rb.

Please review it!

### CODE REVIEW

**Comment by jcastillocano** [comment]
Using ruby 2.X syntax on test file.

**Comment by jcastillocano** [comment]
The behaviour of UDP is more like 'having a socket open on it by a process' than 'I can reach that port because I have a valid response', but I guess it's worthy to have it integreated in check-ports.rb.

File: bin/check-ports.rb [hunk 1]
```diff
@@ -48,6 +48,12 @@ class CheckPort < Sensu::Plugin::Check::CLI
          description: 'Ports to check, comma separated (22,25,8100-8131,3030)',
          default: '22'
 
+  option :proto,
+         short: '-P PROTOCOL',
+         long: '--protocol PROTOCOL',
+         description: 'Protocol to check: tpc (default) or upd',
```
Comment (analytically) [inline bin/check-ports.rb]: Typo: should be udp.

File: test/check-ports_spec.rb [hunk 2]
```diff
@@ -0,0 +1,141 @@
+#! /usr/bin/env ruby
```
Comment (analytically) [inline test/check-ports_spec.rb]: no space

File: bin/check-ports.rb (hunk 1 above)
Comment (jcastillocano) [inline bin/check-ports.rb]: Fixed

Comment (analytically) [inline bin/check-ports.rb]: tpc? mind double checking the PR for any other typos?

File: test/check-ports_spec.rb (hunk 2 above)
Comment (jcastillocano) [inline test/check-ports_spec.rb]: Fixed. Also updated check-mtu_spec.rb

File: bin/check-ports.rb (hunk 1 above)
Comment (jcastillocano) [inline bin/check-ports.rb]: Review both check-ports.rb and check-ports_spec.rb.

**Pull request merged**